src/api/routes/v1/amendments.ts

```diff
@@ -1,54 +1,309 @@
+/* eslint-disable max-lines -- Disable for this file. */
 import { Request, Response } from 'express'
 
-import { query } from '../../../shared/database'
+import { getNetworks, query } from '../../../shared/database'
 import { AmendmentsInfo } from '../../../shared/types'
 import { isEarlierVersion } from '../../../shared/utils'
 import logger from '../../../shared/utils/logger'
 
+import { CACHE_INTERVAL_MILLIS } from './utils'
+
 interface AmendmentsInfoResponse {
   result: 'success' | 'error'
   count: number
   amendments: AmendmentsInfo[]
 }
 
+interface AmendmentsVoteResponse {
+  result: 'success' | 'error'
+  count: number
+  amendments: Array<AmendmentsEnabled | AmendmentInVoting>
+}
 interface SingleAmendmentInfoResponse {
   result: 'success' | 'error'
   amendment: AmendmentsInfo
 }
 
-interface Cache {
+interface CacheInfo {
   amendments: AmendmentsInfo[]
   time: number
 }
 
+interface AmendmentInVoting extends AmendmentsInfo {
+  threshold: string
+  consensus: string
+  voted: {
+    count: number
+    validators: Array<{
```

Comment:
This could become its own type.  Also these two types seem like they have a lot of redundancy.